### PR TITLE
SWIFT-729: Update issue templates

### PR DESCRIPTION
**Pull request by bajanam**
This migrates our legacy issue template to the new template   format, minor clean up.  Pending Further clean up clarifications.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+labels: bug
+assignees: ''
+
+---
+
+<!--
+  Make sure you have read CONTRIBUTING.md completely before you file a new
+  issue! 
+
+  If possible, try to determine if the bug is actually part of the Swift driver,
```
Comment (kmahar) [inline .github/ISSUE_TEMPLATE/bug_report.md:14]: let's remove this line, I think it's a lot easier for us to determine this than for a user.

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  or if the issue is actually from `libmongoc` or `libbson`. If so, you should
+  file the issue with the representative projects.
+-->
+
```
Comment (kmahar) [inline .github/ISSUE_TEMPLATE/bug_report.md:18]: I'd suggest we condense the following items into something shorter, like

1. What version of Swift are you using? (Run `swift --version`)
2. What operating system are you using? (Run `uname -a`)
3. What versions of the driver and its dependencies are you using? (Run `swift package show-dependencies`)
4. What version of MongoDB are you using? (Check with the MongoDB shell using `db.version()`) 
5. What is your MongoDB topology (standalone, replica set, sharded cluster, serverless)?

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+**To Reproduce**
+Steps to reproduce the behavior:
+1. First, do this.
+2. Then do this.
+3. After doing that, do this.
+4. And then, finally, do this.
+5. Bug occurs.
+
+
+**Screenshots**
```
Comment (kmahar) [inline .github/ISSUE_TEMPLATE/bug_report.md:77]: let's remove this. id rather people give us code samples

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+A clear and concise description of what the bug is.
+
+**BE SPECIFIC**:
+* What is the _expected_ behavior and what is _actually_ happening?
+* Do you have any particular output that demonstrates this problem?
+* Do you have any ideas on _why_ this may be happening that could give us a
+clue in the right direction?
+* Did this issue arise out of nowhere, or after an update (of the driver,
+server, and/or Swift)? 
+* Is there a workaround that seems to avoid this this issue?
```
Comment (kmahar) [inline .github/ISSUE_TEMPLATE/bug_report.md:61]: let's remove this and the next bullet, I dont think they are very helpful to us

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+* Are there multiple ways of triggering this bug (perhaps more than one
+function produce a crash)?
+* If you know how to reproduce the bug, use the `BugReport` project in our
```
Comment (kmahar) [inline .github/ISSUE_TEMPLATE/bug_report.md:64]: let's change to "If you know how to reproduce this bug, please include a code snippet here:"

**Review by kmahar — CHANGES_REQUESTED** [review]
some suggestions for things to condense/cut

**Review by bajanam — COMMENTED** [review]
Updates made

**Review by kmahar — APPROVED** [review]
lgtm, thanks! adding in Patrick and Isabel

**Review by isabelatkinson — APPROVED** [review]

**Review by patrickfreed — APPROVED** [review]
LGTM!

**Pull request merged**